src/tensorlake/function_executor/allocation_runner/sdk_algorithms.py

```diff
@@ -454,29 +454,67 @@ def validate_and_deserialize_function_call_metadata(
 
 
 def reconstruct_function_call_args(
+    function: Function,
     function_call_metadata: FunctionCallMetadata | ReduceOperationMetadata | None,
     arg_values: Dict[str, Value],
-    function_instance_arg: Any | None,
 ) -> tuple[List[Any], Dict[str, Any]]:
-    """Returns function call args and kwargs reconstructed from arg_values."""
+    """Returns function call args and kwargs reconstructed from arg_values.
+
+    For class methods, inserts the class instance as the first argument (self).
```

Comment:
This is not true anymore after this change. Also this comment doesn't add much value.